xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp

```diff
@@ -561,30 +561,15 @@ void CGUIDialogPVRTimerSettings::OnSettingAction(const std::shared_ptr<const CSe
 
 bool CGUIDialogPVRTimerSettings::Validate()
 {
-  bool bStartAnyTime = m_bStartAnyTime;
-  bool bEndAnyTime = m_bEndAnyTime;
-
-  if (!m_timerType->SupportsStartAnyTime() ||
-      !m_timerType->IsEpgBased()) // Start anytime toggle is not displayed
-    bStartAnyTime = false; // Assume start time change needs checking for
-
-  if (!m_timerType->SupportsEndAnyTime() ||
-      !m_timerType->IsEpgBased()) // End anytime toggle is not displayed
-    bEndAnyTime = false; // Assume end time change needs checking for
-
-  // Begin and end time
-  if (!bStartAnyTime && !bEndAnyTime)
+  // When timer has a start and end time, compare them but only when it's
+  // not a timer rule, since those only have times not dates
+  if (!m_timerType->IsTimerRule() && !m_bStartAnyTime && !m_bEndAnyTime &&
```

Comment:
the original code would show an error box only for timer rules. Now it is the opposite. Not sure this will actually still fix the original use case from @phunkyfish.

> "not a timer rule, since those only have times not dates"

Need to think about whether this is actually true. I feel it's not.

Reply:
Okay, I adgree that the original code is really hard to understand - maybe my last comment regarding reversed logic regarding timer rule was wrong.

But still your comment regarding timer rules only having times not dates seems not correct.

Reply:
Well I did some more testing here with both normal (epg) timers and (epg) timer rules and it seems for the TVHeadend backend there's no point in validating anything at all. For TVHeadend timer rules at most it can have a start and end time (or start time only), but you can't check those since it doesn't have a date so a start of 23:00 and end of 1:00 shouldn't fail. And for normal timers with a date it's impossible to change the date spiners so that endTimeDate < startTimeDate. So could @phunkyfish please provide a valid use-case where Validate() can catch specific user errors?